jackson-modules/jackson-conversions-2/src/test/java/com/baeldung/jackson/snakecase/SnakeCaseUnitTest.java

```diff
@@ -0,0 +1,55 @@
+package com.baeldung.jackson.snakecase;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.PropertyNamingStrategy;
+import com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class SnakeCaseUnitTest {
+
+    private static final String JSON = "{\"first_name\": \"Jackie\", \"last_name\": \"Chan\"}";
+
+    public static void main(String[] args) throws JsonProcessingException {
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.readValue(JSON, User.class);
+    }
+
+    @Test(expected = UnrecognizedPropertyException.class)
+    public void whenExceptionThrown_thenExpectationSatisfied() throws JsonProcessingException {
```

Comment:
Unit tests should just `throws Exception` rather than a specific one. It's simpler and nobody cares except the test runner, which doesn't have to manage the checked exception in any specific way.